matlabMeas/MeasurementUtilities/initializeSR830Meas2D_Func.m

```diff
@@ -1,73 +1,62 @@
-function [plotHandle] = initializeSR830Meas2D_Func(sweepType1, sweepType1,doBackAndForth)
+function [plotHandle] = initializeSR830Meas2D_Func(sweepTypes, starts, stops, deltas)
 
-[time,Real,Imag] = deal(inf);
+% initializeSR830Meas2D_Func({'Freq', 'ST'}, {1000, 0}, {10000, 1}, {1000, 0.01})
 
-timeLabel = "Time [s]";
-voltageAxisName = genSR830Axis(sweepType);
-yLabel = "Current [A]";
+[sweepType1, sweepType2] = sweepTypes{:};
+[start1, start2] = starts{:};
+[stop1, stop2] = stops{:};
+[deltaParam1, deltaParam2] = deltas{:};
 
-subPlotFigure = figure(getNextMATLABFigNum());
-subplot(2,3,1);
-realVsTime = plotData(time,Real,'xLabel',timeLabel,'yLabel',"Real",'title',"Real vs Time",'subPlot',1);
+xData = [start2, stop2];
+lenX = length(start2:(2 * (start2 < stop2) - 1) * deltaParam2:stop2);
+yData = [start1, stop1];
+lenY = length(start1:(2 * (start1 < stop1) - 1) *deltaParam1:stop1);
+zData = NaN(lenY, lenX);
 
-subplot(2,3,2)
-imagVsTime = plotData(time,Real,'xLabel',timeLabel,'yLabel',"Imag",'title',"Imag vs Time",'subPlot',1);
+yAxisName = genSR830Axis(sweepType1);
+xAxisName = genSR830Axis(sweepType2);
 
-subplot(2,3,3)
-magVsTime = plotData(time,Real,'xLabel',timeLabel,'yLabel',"Mag",'title',"Mag vs Time",'subPlot',1);
+myFig = figure(getNextMATLABFigNum());
 
+fig = image(xData, yData, zData, 'CDataMapping', 'scaled');
 
-
-%[vavg1,vavg2,avgxs1,avgxs2,avgys1,avgys2,stdx1,stdx2,stdy1,stdy2,avgmags1,avgmags2,stdm1,stdm2] = deal(inf);
-
-
-subplot(2,3,4)
-if ~doBackAndForth
-    realVsVoltageErr = errorbar(time,Real,Imag,'Bx');
+if start2 < stop2
+    set(gca,'XDir','normal');
 else
-    realVsVoltageErr = errorbar(time,Real,Imag,'Bx');
-    hold on
-    realVsVoltageErr2 = errorbar(time,Real,Imag,'C*');
-    hold off
+    set(gca,'XDir','reverse');
 end
-xlabel(voltageAxisName);
-ylabel(yLabel);
-title("Real vs Voltage");
 
-subplot(2,3,5)
-if ~doBackAndForth
-    imagVsVoltageErr = errorbar(time,Real,Imag,'RX');
+if start1 < stop1
```

Comment:
I would maybe encapsulate these if statements in a helper function below. It's unclear what this actually does and maybe giving the function a name that people can understand, this piece of code can be clarified.

Reply:
I see. I will add comments and put it in a helper function. These statements help to realign the axes in the direction of sweeping (i.e. if you tell it to start at 2 and sweep down to -1, the leftmost label on the axis will be 2 and the rightmost will be -1. This isn't entirely necessary either, just something I included when coding because it made things easier to debug.